scripts/tcs_startpool.sh

```diff
@@ -0,0 +1,311 @@
+#!/bin/bash
+
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+KV_STORAGE="kv_storage"
+ENCLAVE_MANAGER_KME="${TCF_HOME}/enclave_manager/avalon_enclave_manager/kme/kme_enclave_manager.py"
+ENCLAVE_MANAGER_WPE="${TCF_HOME}/enclave_manager/avalon_enclave_manager/wpe/wpe_enclave_manager.py"
+LISTENER="avalon_listener"
+VERSION="$(cat ${TCF_HOME}/VERSION)"
+
+# Default values
+COMPONENTS="$ENCLAVE_MANAGER_KME $ENCLAVE_MANAGER_WPE" # #KV_STORAGE added if -s passed
+START_STOP_AVALON_SERVICES=0 # default if -s not passed
+LMDB_URL="http://localhost:9090" # -l default
+LISTENER_URL="http://localhost:1947"
+ENCLAVE_ZMQ_URL="tcp://localhost:5555"
+
+#Variables
+COUNT=1
+PORT=1948
+WORKER_ID=0
+declare -A WORKERPOOL
+
+is_sync_mode()
+{
+    return grep "sync_workload_execution" ${TCF_HOME}/listener/listener_config.toml | awk -F'=' '{print $2}'
+}
+
+start_avalon_components()
+{
+    if [ $START_STOP_AVALON_SERVICES = 1 ] ; then
+    echo "Starting Avalon KV Storage $VERSION ..."
+    $KV_STORAGE --bind $LMDB_URL & 
+    echo "Avalon KV Storage started"
+    fi
+
+    if [ $START_STOP_AVALON_SERVICES = 1 ] ; then
+        echo "Starting Avalon Listener $VERSION ..."
+        is_sync_mode
+        is_sync_mode_on=$?
```

Comment:
why to use is_sync_mode_on when there is is_sync_mode?

Reply:
It is shell convention to call function. @51 function call to is_sync_mode and @52 is variable name to capture the return value of function is_sync_mode.